deployment/build_and_release/modules/release/main.tf

```
@@ -235,4 +235,232 @@ resource "google_cloudbuild_trigger" "release" {
   filename = "${each.value.cloudbuild_path}"
 }
 
+resource "google_cloudbuild_trigger" "os_build" {
+  name = "os-build-${var.env}"
+  location = "global"
+
+  github {
+    owner = "transparency-dev"
+    name  = "armored-witness-os"
+
+    push {
+      branch = var.cloudbuild_trigger_branch != "" ? var.cloudbuild_trigger_branch : null
+      tag = var.cloudbuilgit add d_trigger_tag != "" ? var.cloudbuild_trigger_tag : null
+    }
+  }
+
+  build {
+    # If the trigger is not based on `tag`, create a fake one.
+    #
+    # Unfortunately, GCB has no concept of dynamically creating substitutions or
+    # passing ENV vars between steps, so the best we can do is to create a file
+    # containing our tag in the shared workspace which other steps can inspect.
+    step {
+      name = "bash"
+      script = (
+        var.cloudbuild_trigger_tag != "" ?
+        "$TAG_NAME > /workspace/git_tag && cat /workspace/git_tag" :
+        "date +'0.3.%s-incompatible' > /workspace/git_tag && cat /workspace/git_tag"
+      )
+    }
+    ### Build the Trusted OS and upload it to GCS.
+    # Build an image containing the Trusted OS artifacts with the Dockerfile.
+    # This step needs to be a bash script in order to read the tag content from file.
+    step {
+      name = "gcr.io/cloud-builders/docker"
+      entrypoint = "bash"
+      args = [
+        "-c",
+        <<EOT
```

```suggestion
        <<-EOT
```

Something to consider. This will strip off the indents in a "clever" way. i.e. `docker build` will not have whitespace in front of it, and the same amount of whitespace will be stripped from all the other lines in this block.

Similarly throughout the rest of this file, if you choose to do this.

---

Did this here and in the other places.